src/nupic/algorithms/Connections.hpp

```diff
@@ -322,7 +324,7 @@ namespace nupic
          *
          * @retval Segment data.
          */
-        SegmentData dataForSegment(const Segment& segment) const;
+        SegmentData& dataForSegment(const Segment& segment);
```

Comment:
So are these functions intended for inspecting the state or for modifying it? If the former then it should be const (and return const references). If the latter, then the change makes sense but it changes the semantics of the function and should be documented.